api/server/server.go

```diff
@@ -1091,6 +1091,179 @@ func postContainersCopy(eng *engine.Engine, version version.Version, w http.Resp
 	return nil
 }
 
+func handleCopyError(w http.ResponseWriter, statusp *int, errp *error) {
```

Comment:
I remember looking at the existing `httpError()` and deciding to avoid using it. It even has a comment saying something similar:

    FIXME: this is brittle and should not be necessary.

But for the sake of being consistent I'll rework the new handlers to return errors that `httpError()` can correctly handle.